<a href="https://colab.research.google.com/github/MadisonJBollinger/lis5693/blob/main/5693_lab2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [47]:
!pip install -U pip setuptools wheel
!pip install -U spacy
!python -m spacy download en_core_web_sm

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.8/12.8 MB 22.7 MB/s  0:00:00
✔ Download and installation successful
You can now load the package via spacy.load('en_core_web_sm')
⚠ Restart to reload dependencies
If you are in a Jupyter or Colab notebook, you may need to restart Python in
order to load all the package's dependencies. You can do this by selecting the
'Restart kernel' or 'Restart runtime' option.


In [48]:
import spacy

In [49]:
nlp = spacy.load ("en_core_web_sm")

In [50]:
import requests

url = "https://raw.githubusercontent.com/MadisonJBollinger/lis5693/refs/heads/main/lab-1/transcript.txt"
response = requests.get(url)
response.raise_for_status() # Raise an exception for HTTP errors
text = response.text

In [51]:
print("Number of characters:", len(text)) # print number of characters

print(text[:100])   # print first 100 characters

Number of characters: 13052
[Music]
were roman houses really dank stuffy and
hot all the time
or do they have air conditioning
w


In [52]:
nlp = spacy.blank("en")
nlp.add_pipe("sentencizer")

doc = nlp(text)

sentences = list(doc.sents)

print("Number of sentences:", len(sentences))

print("\nFirst 5 sentences:")
for sent in sentences[:5]:
    print(sent)


Number of sentences: 1

First 5 sentences:
[Music]
were roman houses really dank stuffy and
hot all the time
or do they have air conditioning
what about an attached garage for the
carts and wagons
how big were their front yards and
backyards too
do they all have a great pull about a
hot tub
these and other questions will be
answered right after this
[Music]
ah
[Music]
i'm professor jerome markenberg and i've
been teaching a wide variety of history
courses at colleges across this country
for the past 30 years
in this video i'm going to tell you
about
the roman domis or house
who lived in the single family domi
which is the plural of domes
how these
domi were laid out
and the various rooms in a roman domis
their name
and function
at the end i'll have the wrap-up quote
on this video but first
make sure to click like share
especially subscribe
and that little bell thingy
so continue to bring you more great
videos
just like this one
a single family house or domus or domi
as i said which is 

Task 3: My transcript seems to not have transcribed correctly as it only show that my transcript is one sentence.

In [53]:
words = [token.text.lower() for token in doc if token.is_alpha]

print("Total words:", len(words))
print("Unique words:", len(set(words)))

Total words: 2449
Unique words: 706


Task 4:There are 2449 words, of which there are 706 unique ones.

In [54]:
from collections import Counter

word_freq = Counter(words)

print("Top 10 most frequent words:")
for word, count in word_freq.most_common(10):
    print(word, count)

Top 10 most frequent words:
the 165
a 67
and 65
here 58
in 49
of 47
to 46
it 45
or 43
you 43


Task 5: "The" is the most frequent word in the transcript, it is a determinator and one of the most common words in the english language. It specifies one item or group of items.

In [58]:
nlp2 = spacy.load("en_core_web_sm")

doc2 = nlp2(text)

print("Named Entities:", len(doc2.ents)) #len tells me the number for it.
for ent in doc2.ents:
    print(ent.text, ent.label_)

Named Entities: 47
jerome markenberg PERSON
the past 30 years DATE
first ORDINAL
romans NORP
paul getty PERSON
malibu california FAC
four CARDINAL
six CARDINAL
romans NORP
greek NORP
the greek house ORG
two CARDINAL
two CARDINAL
greek NORP
today DATE
european NORP
american house ORG
the greek house ORG
rome GPE
winter DATE
summer months DATE
12 to 18 feet QUANTITY
daily DATE
romans NORP
italian NORP
second ORDINAL
second ORDINAL
three CARDINAL
two CARDINAL
more than one CARDINAL
20 CARDINAL
10 CARDINAL
greek NORP
romans NORP
kulina GPE
summer DATE
two CARDINAL
greek NORP
romans NORP
dutch NORP
night TIME
one CARDINAL
four CARDINAL
four CARDINAL
20 CARDINAL
two CARDINAL
50 CARDINAL


Task 6: There were 47 entites found, I used len to give me the number instead of counting manually. PERSON,DATE,ORDINAL, NORP,FAC, CARDINAL, DATE, GPE, ORG, QUANTITY, DATE, and TIME were all the types found in the transcript.

In [59]:
from spacy.matcher import PhraseMatcher

matcher = PhraseMatcher(nlp2.vocab, attr="LOWER")

phrases = ["Rome", "Room", "Community"]

patterns = [nlp2(p) for p in phrases]

matcher.add("TECH_TERMS", patterns)

matches = matcher(doc2)

print("Matches found:")
for match_id, start, end in matches:
    print(doc2[start:end])
    print("Sentence:", doc2[start].sent)

Matches found:
room
Sentence: well you
don't kick it
you're not trying to smash it in you
sort of knock on it
it would be easier to knock it out with
your hand but instead they use their
feet
once past the door let in by the
guardian slave
you would find yourself in an entrance
hall or
vestibulum let me see the vestibulum
here
these would lead directly into the focal
point of the house
the atrium
as i said
unlike the greek house the roman house
did not have an andron
a room where solely the men of the house
could meet talk dine and party with men
not part of the family
where the only woman allowed would have
been dancers entertainers or prostitutes
and not the women of the family
largely because rome is described
elsewhere in another video
was not nearly as patriarchal
as greece they were patriarchal but not
nearly as bad
and skywell style domi the atrium again
shown here
various depictions
was a location of the important skywell
which allowed existing sunlight into the
house
while als

I chose a video on Roman housing. It is not only a personal interest of mine, but we also were discussing housing in Rome in the youth bible study that I taught last year, and it made me even more fascinated.

Reflection:
The tasks themselves were easy to navigate. Following the practice with the spaCy, the assignment itself was a breeze. However, I started the assignment with a new transcript file before realising that I misread the assignment and had to redo it to match the Lab 1 transcrip. What was difficult though was that I mention earlier in the coding, the transcript only shows one sentence, which makes it more wordy in the output making it harder to find the items requested to complete the assignment.